PolyEngine/Engine/Src/Engine.hpp

```diff
@@ -127,60 +87,6 @@ namespace Poly
 		/// <returns>Pointer to current world.</returns>
 		World* GetWorld() { return BaseWorld.get(); }
 
-		/// <summary>Registers component tyoe for further use.
-		/// Registered class must inherit from ComponentBase class.</summary>
-		/// <tparam name="T">component typen</tparam>
-		/// <param name="id">Specifies what id should be associated to registered component.</param>
-		template<typename T> void RegisterComponent(size_t id)
-		{
-			ASSERTE(
-				([this, &id] () ->bool
-				{
-					for(auto it = ComponentTypeMap.begin(); it != ComponentTypeMap.end(); ++it)
-						if(it->first == typeid(T) || it->second == id) return false;
-					return true;
-				})(),
-				"Component type or id was registered twice!");
-
-			ComponentTypeMap[typeid(T)] = id;
-		}
-
-		/// <summary>If given component is registered function returns associated ID.</summary>
-		/// <tparam name="T">Typename which ID is requested</tparam>
-		/// <returns>Associated ID.</returns>
-		template<typename T> size_t GetComponentID() const
```

Comment:
It would be nicer to use this method instead of the one in Family for getting component ID. The we can hide Family behind `Impl` namespace.

Reply:
Moving this method from Engine.hpp to World.hpp saves us one pointer call (i.e. abandon referencing via gEngine-> in World.hpp).
Apart from that, in World.hpp we create the division into World/'Regular' components (AddComponent(), AddWorldComponent()) so the division of Family naturally fits in there.

Reply:
done. created static method GetComponentID() in World.hpp